tools/cherry-picker/src/main.ts

```diff
@@ -0,0 +1,156 @@
+    );
+    return;
+  }
+
+  // Find the commits on the master
+  const commitsOnMaster = await findMergedCommitsOnMaster(client, prNumber);
+
+  // We need to reverse the found commits, to keep them in the right order
+  // when cherry picking.
+  const reversedCommitsOnMaster = commitsOnMaster.reverse();
+
+  // If we were not able to map the commits from the pull request to master
+  // commits
+  if (!commitsOnMaster.length) {
+    await pullRequestAddComment(
+      client,
+      prNumber,
+      '@dynatrace-oss/barista The auto cherry picker failed to determine the commits on master, that should be cherry picked.',
+    );
+    throw new Error('Not able to map pull request commit to master commit');
+  }
+
+  // Get the current minor and patch branches based on the version of the
+  // master package json.
+  const {
+    currentMinorBranch,
+    currentPatchBranch,
+  } = await getCurrentMinorAndPatchBranches();
+
+  const cherryPickMessages: string[] = [];
+  if (pullRequestHasLabel(pullRequestDetails, 'target: minor')) {
+    try {
+      // Cherry pick to the minor branch
+      const cherryPickToMinor = await cherryPickCommits({
+        commits: reversedCommitsOnMaster,
+        head: currentMinorBranch,
+        octokit: client,
+        owner: context.repo.owner,
+        repo: context.repo.repo,
+      });
+      cherryPickMessages.push(
+        `Cherry picked to minor branch ${currentMinorBranch}. New head is now ${cherryPickToMinor}`,
+      );
+    } catch (err) {
+      await pullRequestAddComment(
+        client,
+        prNumber,
+        '@dynatrace-oss/barista, the auto cherry picker failed. Please have a look at the failure and cherry pick by hand.',
+      );
+      console.log(
+        `Failed to cherry pick ${reversedCommitsOnMaster.join(
+          ', ',
+        )} to ${currentMinorBranch}.`,
+      );
+      throw err;
+    }
+  }
+
+  if (pullRequestHasLabel(pullRequestDetails, 'target: patch')) {
+    try {
+      // Cherry pick to the minor branch
+      const cherryPickToPatch = await cherryPickCommits({
+        commits: reversedCommitsOnMaster,
+        head: currentPatchBranch,
+        octokit: client,
+        owner: context.repo.owner,
+        repo: context.repo.repo,
+      });
+      cherryPickMessages.push(
+        `Cherry picked to patch branch ${currentPatchBranch}. New head is now ${cherryPickToPatch}`,
+      );
+    } catch (err) {
+      await pullRequestAddComment(
+        client,
+        prNumber,
+        '@dynatrace-oss/barista, the auto cherry picker failed. Please have a look at the failure and cherry pick by hand.',
+      );
+      console.log(
+        `Failed to cherry pick ${reversedCommitsOnMaster.join(
+          ', ',
+        )} to ${currentPatchBranch}.`,
+      );
+      throw err;
+    }
+  }
+  await pullRequestAddComment(
+    client,
+    prNumber,
+    `The auto cherry picker succeeded.${cherryPickMessages.join('\n')}`,
+  );
+}
+
+run();
```

Comment:
please handle the promise and catch errors.

```suggestion
run().then(() => {}).catch((error) => console.log(error));
```